src/sentry/grouping/strategies/hierarchical.py

```diff
@@ -0,0 +1,155 @@
+from sentry.grouping.component import GroupingComponent
+from sentry.utils.safe import get_path
+
+MAX_LAYERS = 5
+
+
+def get_stacktrace_hierarchy(main_variant, components, frames, inverted_hierarchy):
+    main_variant.update(tree_label="<entire stacktrace>")
+
+    frames_iter = list(zip(frames, components))
+    if not inverted_hierarchy:
+        frames_iter = reversed(frames_iter)
```

Comment:
A comment might be nice here about why the frames are reversed when the flag is `False` and not the other way round.